theano/tests/unittest_tools.py

```diff
@@ -244,9 +244,9 @@ def _compile_and_check(self, inputs, outputs, numeric_inputs, cls,
         # Check that the Op is removed from the compiled function.
         if check_topo:
             topo_shape = shapes_function.maker.fgraph.toposort()
+            topo_out = outputs_function.maker.fgraph.toposort()
             assert not any(isinstance(t.op, cls) for t in topo_shape)
-        topo_out = outputs_function.maker.fgraph.toposort()
-        assert any(isinstance(t.op, cls) for t in topo_out)
```

Comment:
Why this change? I don't think it is the right fix, we want the old behavior most of the time. If you really need that, add a new parameter.